### PR TITLE
Rename version to api_version

### LINKED ISSUE #1199: Add api_version to TenantConfiguration

### PR DESCRIPTION
**Pull request by louischan-oursky**
fix #1199

### CODE REVIEW

File: pkg/core/middleware/apiversion_test.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+					"code": 404,
+					"message": "incompatible API version",
+					"name": "NotFound",
+					"reason": "IncompatibleAPIVersion"
+				}
+			}
+			`)
+		}
+
+		r, _ := http.NewRequest("GET", "/v3.1/foobar", nil)
```
Comment (carmenlau) [inline pkg/core/middleware/apiversion_test.go]: Please add one more test for given minor version less than the server minor version

File: pkg/core/server/server.go [hunk 2]
```diff
@@ -41,14 +41,18 @@ func NewServerWithOption(
...
+		router = router.PathPrefix(option.GearPathPrefix).Subrouter()
 	}
+
+	if option.IsAPIVersioned {
+		router = router.PathPrefix("/{api_version}").Subrouter()
+	}
+
 	srv := Server{
-		router: subRouter,
+		router: router,
```
Comment (carmenlau) [inline pkg/core/server/server.go]: The `/healthz` check endpoint become 404 after this change

File: pkg/core/apiversion/apiversion.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	"fmt"
+	"regexp"
+	"strconv"
+)
+
+// MajorVersion is the current major API Version.
+const MajorVersion = 2
+
+// MinorVersion is the current minor API Version.
+const MinorVersion = 0
```
Comment (kiootic) [inline pkg/core/apiversion/apiversion.go]: So we start from 2.0? I was thinking of following the milestone version (i.e. currently 2.1) for easier tracking.

File: pkg/core/config/schema.go [hunk 4]
```diff
@@ -24,14 +24,14 @@ const (
 		"$id": "#TenantConfiguration",
 		"type": "object",
 		"properties": {
-			"version": { "const": "2" },
+			"api_version": { "enum": ["v2.0"] },
```
Comment (kiootic) [inline pkg/core/config/schema.go]: Can we use the version string defined in `apiversion`? It seems quite easy to miss when updating the version.

File: pkg/core/middleware/apiversion.go [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+func (m *APIVersionMiddleware) Handle(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		apiVersion := mux.Vars(r)[m.APIVersionName]
+		major, minor, ok := apiversion.Parse(apiVersion)
+
+		if !ok || major != m.MajorVersion || minor > m.MinorVersion {
+			handler.WriteResponse(w, handler.APIResponse{
+				Error: ErrIncompatibleAPIVersion,
```
Comment (kiootic) [inline pkg/core/middleware/apiversion.go]: We may want to include the expected version, so the clients know which version they should use.

**Comment by louischan-oursky** [comment]
comments addressed.

**Pull request merged**